File: common-lib/src/main/java/edu/iuh/exchange/common/exception/BaseException.java
package edu.iuh.exchange.common.exception;

import org.springframework.http.HttpStatus;

/**
 * Base exception cho toàn bộ microservices.
 * Mọi business exception đều extend từ class này.
 */
public class BaseException extends RuntimeException {

    private final HttpStatus status;
    private final String errorCode;

    public BaseException(HttpStatus status, String errorCode, String message) {
        super(message);
        this.status = status;
        this.errorCode = errorCode;
    }

    public HttpStatus getStatus() {
        return status;
    }

    public String getErrorCode() {
        return errorCode;
    }
}

File: node-services/packages/common/src/exceptions/BaseException.js
/**
 * Base exception cho toàn bộ microservices.
 * Mọi business exception đều extend từ class này.
 */
export class BaseException extends Error {
  /**
   * @param {number} status - HTTP status code
   * @param {string} errorCode - Business error code
   * @param {string} message - Human-readable message
   */
  constructor(status, errorCode, message) {
    super(message);
    this.name = this.constructor.name;
    this.status = status;
    this.errorCode = errorCode;
    Error.captureStackTrace(this, this.constructor);
  }
}
